Assets/MixedRealityToolkit.SDK/Experimental/Features/UX/MixedRealityKeyboardBase.cs

```diff
@@ -0,0 +1,160 @@
+// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT License. See LICENSE in the project root for license information.
+
+using UnityEngine;
+using UnityEngine.UI;
+#if WINDOWS_UWP
+using Windows.UI.ViewManagement;
+using Microsoft.MixedReality.Toolkit.Input;
+using System.Collections;
+#endif
+
+namespace Microsoft.MixedReality.Toolkit.Experimental.UI
+{
+    /// <summary>
+    /// Class that can launch and hide a system keyboard specifically for Windows Mixed Reality
+    /// devices (HoloLens 2, Windows Mixed Reality).
+    /// 
+    /// Implements a workaround for UWP TouchScreenKeyboard bug which prevents
+    /// UWP keyboard from showing up again after it is closed.
+    /// Unity bug tracking the issue https://fogbugz.unity3d.com/default.asp?1137074_rttdnt8t1lccmtd3
+    /// </summary>
+    public abstract class MixedRealityKeyboardBase<T> : MonoBehaviour where T : Selectable
+    {
+#if WINDOWS_UWP
+        protected T inputField;
+
+        #region Properties
+
+        public bool Visible => State == KeyboardState.Showing;
+
+        #endregion properties
+
+        #region Private fields
+
+        private InputPane inputPane = null;
+        private TouchScreenKeyboard keyboard = null;
+
+        private KeyboardState State = KeyboardState.Hidden;
+
+        private Coroutine stateUpdate;
+
+        #endregion private fields
+
+        #region Private enums
+
+        private enum KeyboardState
+        {
+            Hiding,
+            Hidden,
+            Showing,
+        }
+
+        #endregion Private enums
+
+        #region Unity functions
+
+        private void Awake()
+        {
+            if ((inputField = GetComponent<T>()) == null)
+            {
+                Destroy(this);
```

Comment:
Destroy seems a little aggressive....regardless add console error/warning or something to make it clear to user in logs

Reply:
You're right, I could just remove it. Both deriving classes use RequireComponent and this one is abstract.

Reply:
Regardless of removing it, there should be a logging line so people know it has failed